Dockerfile

```diff
@@ -2,7 +2,8 @@ FROM openjdk:8-alpine
 
 # build and cleanup
 COPY setup.sh /opt/setup.sh
-RUN chmod +x /opt/setup.sh && /opt/setup.sh
```

Comment:
Running the chained command on linux:
```
$ uname -vr
4.4.0-64-generic #85-Ubuntu SMP Mon Feb 20 11:50:30 UTC 2017
$ docker -v
Docker version 17.03.1-ce, build c6d412
$ docker build -t test .
Sending build context to Docker daemon 105.5 kB
Step 1/6 : FROM openjdk:8-alpine
 ---> 478bf389b75b
Step 2/6 : COPY setup.sh /opt/setup.sh
 ---> b8cd5c1fb84f
Removing intermediate container ac53165aa10f
Step 3/6 : RUN chmod +x /opt/setup.sh && /opt/setup.sh
 ---> Running in 888f76751718
/bin/sh: /opt/setup.sh: Text file busy
The command '/bin/sh -c chmod +x /opt/setup.sh && /opt/setup.sh' returned a non-zero code: 2
```

Running it on OSX works as intended both ways, so splitting it into two lines is a more cross platform solution. And fortunately it doesn't add (much, as in less than a KB) to the final image size.